include/h2o.h

```diff
@@ -2551,6 +2569,17 @@ inline int h2o_req_can_stream_request(h2o_req_t *req)
     return first_handler != NULL && first_handler->supports_request_streaming;
 }
 
+inline int h2o_req_should_forward_expect(h2o_req_t *req)
+{
+    h2o_handler_t **handlers;
+    size_t size = h2o_get_handlers(req, &handlers);
+    for (size_t i = 0; i != size; ++i) {
+        if (handlers[i]->supports_expect)
+            return 1;
+    }
+    return 0;
```

Comment:
I think we'd only want to check the first handler, like `h2o_req_can_stream_request`.

Consider the case where the first handler does not support `supports_expect` but the second handler does. The first handler might block until all the request body arrives. To let that happen, the protocol handlers need to send 100 themselves.

Reply:
Addressed in https://github.com/h2o/h2o/pull/3419/commits/71e29f72f07d4dfdcff1adbb073e33c3bdf19335